src/gui/qgsmaptoolidentify.cpp

```diff
@@ -1054,6 +1054,20 @@ bool QgsMapToolIdentify::identifyRasterLayer( QList<IdentifyResult> *results, Qg
   derivedAttributes.insert( derivedAttributesForPoint( QgsPoint( pointInCanvasCrs ) ) );
 #endif
 
+  // Get raster row and column of clicked point
+  // Thanks to the Value Tool plugin developers for the math
+  const QgsRectangle extent = dprovider->extent();
+  const int rasterRow = std::floor( ( extent.yMaximum() - point.y() ) / layer->rasterUnitsPerPixelY() );
```

Comment:
It's possible here that rasterUnitsPerPixel will be 0 (eg for XYZ or other rasters where the concept doesn't make sense). 

We should check this here, and only add the pixel coordinates if it's non zero

Reply:
Good call! I added a check.

https://github.com/qgis/QGIS/blob/66fa7e7552a762ae56bdfee1bd42f839db037cf2/src/core/raster/qgsrasterdataprovider.cpp#L305 has the same issue then I think.